config/routes.rb

```diff
@@ -58,7 +58,7 @@
     PROJECT_CONSTRAINT = /[^\/]+/
     VERSION_CONSTRAINT = /[\w\.\-]+/
 
-    put '/maintenance/stats/enqueue/:platform/:name', as: :maintenance_stat_enqueue, to: 'maintenance_stats#enqueue', constraints: { :platform => PLATFORM_CONSTRAINT, :name => PROJECT_CONSTRAINT }
+    put '/maintenance/stats/enqueue/:platform/:name', as: :maintenance_stat_enqueue, to: 'maintenance_stats#enqueue', constraints: { :platform => PLATFORM_CONSTRAINT, :name => /.*/ }
     get '/maintenance/stats/begin/:platform/:name', to: 'maintenance_stats#begin_watching', constraints: { :platform => PLATFORM_CONSTRAINT, :name => PROJECT_CONSTRAINT }
```

Comment:
I don't know if these other routes should have the same constraint. There doesn't seem to be a clear pattern in the routes to me.

Reply:
Should we instead be url encoding the name when creating this link?

Reply:
I added encoding for that param and decoding in the `application_controller`.  I don't know how I feel about putting the decoding in such a fundamental method. The other option is to create something similar to what is in place for Project sync instead of embedding an API call in the frontend.

Reply:
The little route that doesn't require the encoding and isn't an api route sounds even better :)